File: Engine/Math/Rect.cpp
#include "Rect.hpp"

Vec2 Rect::GetCenter() const {
	return Pos + Size * 0.5f;
}
float Rect::GetTop() const {
	return Pos.y;
}
float Rect::GetBottom() const {
	return Pos.y + Size.y;
}
float Rect::GetLeft() const {
	return Pos.x;
}
float Rect::GetRight() const {
	return Pos.x + Size.x;
}

File: MonoScripting/Api/Rect.cs
public struct Rect
   {
    public Rect(Vec2 pos, Vec2 size)
    {
        this.Pos = pos;
        this.Size = size;
    }

    public Rect(float posX, float posY, float width, float height){
        this.Pos.x = posX;
        this.Pos.y = posY;
        this.Size.x = width;
        this.Size.y = height;
    }

    public Vec2 Pos;
    public Vec2 Size;
}
